ext/TestExt.jl

```diff
@@ -88,11 +88,15 @@ function TranscodingStreams.test_chunked_read(Encoder, Decoder)
         ok = true
         for chunk in chunks
             stream = TranscodingStream(Decoder(), buffer, stop_on_end=true)
-            ok &= hash(read(stream)) == hash(chunk)
+            ok &= read(stream) == chunk
             ok &= eof(stream)
             ok &= isreadable(stream)
             close(stream)
         end
+        # read without stop_on_end should read the full data.
```

Comment:
This test isn't related to this PR but was something I noticed wasn't covered by current tests.